src/app/data.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import {Post} from "./post";
+import {HttpClient, HttpHeaders, HttpParams} from "@angular/common/http";
+import {createI18nOptions} from "@angular-devkit/build-angular/src/utils/i18n-options";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DataService {
+
+  posts!: Post[];
```

Comment:
Why do you use `!` operator here?

Reply:
Because variable initially null or undefind